diff --git a/node_modules/@apollo/react-hoc/lib/graphql.d.ts b/node_modules/@apollo/react-hoc/lib/graphql.d.ts
index bfe841f..6b2f338 100644
--- a/node_modules/@apollo/react-hoc/lib/graphql.d.ts
+++ b/node_modules/@apollo/react-hoc/lib/graphql.d.ts
@@ -1,4 +1,5 @@
 /// <reference types="react" />
 import { DocumentNode } from 'graphql';
+import { TypedDocumentNode } from "@graphql-typed-document-node/core";
 import { OperationOption, DataProps, MutateProps } from './types';
-export declare function graphql<TProps extends TGraphQLVariables | {} = {}, TData = {}, TGraphQLVariables = {}, TChildProps = Partial<DataProps<TData, TGraphQLVariables>> & Partial<MutateProps<TData, TGraphQLVariables>>>(document: DocumentNode, operationOptions?: OperationOption<TProps, TData, TGraphQLVariables, TChildProps>): (WrappedComponent: import("react").ComponentType<TProps & TChildProps>) => import("react").ComponentClass<TProps, any>;
+export declare function graphql<TProps extends TGraphQLVariables | {} = {}, TData = {}, TGraphQLVariables = {}, TChildProps = Partial<DataProps<TData, TGraphQLVariables>> & Partial<MutateProps<TData, TGraphQLVariables>>>(document: DocumentNode | TypedDocumentNode<TData, TGraphQLVariables>, operationOptions?: OperationOption<TProps, TData, TGraphQLVariables, TChildProps>): (WrappedComponent: import("react").ComponentType<TProps & TChildProps>) => import("react").ComponentClass<TProps, any>;
diff --git a/node_modules/@apollo/react-hoc/lib/mutation-hoc.d.ts b/node_modules/@apollo/react-hoc/lib/mutation-hoc.d.ts
index 677112a..afaf199 100644
--- a/node_modules/@apollo/react-hoc/lib/mutation-hoc.d.ts
+++ b/node_modules/@apollo/react-hoc/lib/mutation-hoc.d.ts
@@ -1,4 +1,5 @@
 import React from 'react';
 import { DocumentNode } from 'graphql';
 import { OperationOption, MutateProps } from './types';
-export declare function withMutation<TProps extends TGraphQLVariables | {} = {}, TData = {}, TGraphQLVariables = {}, TChildProps = MutateProps<TData, TGraphQLVariables>>(document: DocumentNode, operationOptions?: OperationOption<TProps, TData, TGraphQLVariables, TChildProps>): (WrappedComponent: React.ComponentType<TProps & TChildProps>) => React.ComponentClass<TProps, any>;
+import { TypedDocumentNode } from "@graphql-typed-document-node/core";
+export declare function withMutation<TProps extends TGraphQLVariables | {} = {}, TData = {}, TGraphQLVariables = {}, TChildProps = MutateProps<TData, TGraphQLVariables>>(document: DocumentNode | TypedDocumentNode<TData, TGraphQLVariables>, operationOptions?: OperationOption<TProps, TData, TGraphQLVariables, TChildProps>): (WrappedComponent: React.ComponentType<TProps & TChildProps>) => React.ComponentClass<TProps, any>;
diff --git a/node_modules/@apollo/react-hoc/lib/query-hoc.d.ts b/node_modules/@apollo/react-hoc/lib/query-hoc.d.ts
index 352e9d0..d5aca0a 100644
--- a/node_modules/@apollo/react-hoc/lib/query-hoc.d.ts
+++ b/node_modules/@apollo/react-hoc/lib/query-hoc.d.ts
@@ -1,4 +1,5 @@
 import React from 'react';
 import { DocumentNode } from 'graphql';
 import { OperationOption, DataProps } from './types';
-export declare function withQuery<TProps extends TGraphQLVariables | {} = {}, TData = {}, TGraphQLVariables = {}, TChildProps = DataProps<TData, TGraphQLVariables>>(document: DocumentNode, operationOptions?: OperationOption<TProps, TData, TGraphQLVariables, TChildProps>): (WrappedComponent: React.ComponentType<TProps & TChildProps>) => React.ComponentClass<TProps, any>;
+import { TypedDocumentNode } from "@graphql-typed-document-node/core";
+export declare function withQuery<TProps extends TGraphQLVariables | {} = {}, TData = {}, TGraphQLVariables = {}, TChildProps = DataProps<TData, TGraphQLVariables>>(document: DocumentNode | TypedDocumentNode<TData, TGraphQLVariables>, operationOptions?: OperationOption<TProps, TData, TGraphQLVariables, TChildProps>): (WrappedComponent: React.ComponentType<TProps & TChildProps>) => React.ComponentClass<TProps, any>;
diff --git a/node_modules/@apollo/react-hoc/lib/subscription-hoc.d.ts b/node_modules/@apollo/react-hoc/lib/subscription-hoc.d.ts
index 42ddc48..820e8fb 100644
--- a/node_modules/@apollo/react-hoc/lib/subscription-hoc.d.ts
+++ b/node_modules/@apollo/react-hoc/lib/subscription-hoc.d.ts
@@ -1,4 +1,5 @@
 import React from 'react';
 import { DocumentNode } from 'graphql';
 import { OperationOption, DataProps } from './types';
-export declare function withSubscription<TProps extends TGraphQLVariables | {} = {}, TData = {}, TGraphQLVariables = {}, TChildProps = DataProps<TData, TGraphQLVariables>>(document: DocumentNode, operationOptions?: OperationOption<TProps, TData, TGraphQLVariables, TChildProps>): (WrappedComponent: React.ComponentType<TProps & TChildProps>) => React.ComponentClass<TProps, any>;
+import { TypedDocumentNode } from "@graphql-typed-document-node/core";
+export declare function withSubscription<TProps extends TGraphQLVariables | {} = {}, TData = {}, TGraphQLVariables = {}, TChildProps = DataProps<TData, TGraphQLVariables>>(document: DocumentNode | TypedDocumentNode<TData, TGraphQLVariables>, operationOptions?: OperationOption<TProps, TData, TGraphQLVariables, TChildProps>): (WrappedComponent: React.ComponentType<TProps & TChildProps>) => React.ComponentClass<TProps, any>;
