diff --git a/node_modules/@apollo/react-components/lib/types.d.ts b/node_modules/@apollo/react-components/lib/types.d.ts
index 0c82d57..d9ceb13 100644
--- a/node_modules/@apollo/react-components/lib/types.d.ts
+++ b/node_modules/@apollo/react-components/lib/types.d.ts
@@ -1,16 +1,17 @@
 /// <reference types="react" />
+import { TypedDocumentNode } from "@graphql-typed-document-node/core";
 import { OperationVariables, QueryFunctionOptions, QueryResult, BaseMutationOptions, MutationFunction, MutationResult, SubscriptionResult } from '@apollo/react-common';
 import { DocumentNode } from 'graphql';
 import { BaseSubscriptionOptions } from '@apollo/react-common';
 export interface QueryComponentOptions<TData = any, TVariables = OperationVariables> extends QueryFunctionOptions<TData, TVariables> {
     children: (result: QueryResult<TData, TVariables>) => JSX.Element | null;
-    query: DocumentNode;
+    query: DocumentNode | TypedDocumentNode<TData, TVariables>;
 }
 export interface MutationComponentOptions<TData = any, TVariables = OperationVariables> extends BaseMutationOptions<TData, TVariables> {
-    mutation: DocumentNode;
+    mutation: DocumentNode | TypedDocumentNode<TData, TVariables>;
     children: (mutateFunction: MutationFunction<TData, TVariables>, result: MutationResult<TData>) => JSX.Element | null;
 }
 export interface SubscriptionComponentOptions<TData = any, TVariables = OperationVariables> extends BaseSubscriptionOptions<TData, TVariables> {
-    subscription: DocumentNode;
+    subscription: DocumentNode | TypedDocumentNode<TData, TVariables>;
     children?: null | ((result: SubscriptionResult<TData>) => JSX.Element | null);
 }
