diff --git a/node_modules/apollo-cache/lib/cache.d.ts b/node_modules/apollo-cache/lib/cache.d.ts
index 7d2c3c3..e04213b 100644
--- a/node_modules/apollo-cache/lib/cache.d.ts
+++ b/node_modules/apollo-cache/lib/cache.d.ts
@@ -2,7 +2,7 @@ import { DocumentNode } from 'graphql';
 import { DataProxy, Cache } from './types';
 export declare type Transaction<T> = (c: ApolloCache<T>) => void;
 export declare abstract class ApolloCache<TSerialized> implements DataProxy {
-    abstract read<T, TVariables = any>(query: Cache.ReadOptions<TVariables>): T | null;
+    abstract read<T, TVariables = any>(query: Cache.ReadOptions<TVariables, T>): T | null;
     abstract write<TResult = any, TVariables = any>(write: Cache.WriteOptions<TResult, TVariables>): void;
     abstract diff<T>(query: Cache.DiffOptions): Cache.DiffResult<T>;
     abstract watch(watch: Cache.WatchOptions): () => void;
@@ -15,8 +15,8 @@ export declare abstract class ApolloCache<TSerialized> implements DataProxy {
     abstract recordOptimisticTransaction(transaction: Transaction<TSerialized>, id: string): void;
     transformDocument(document: DocumentNode): DocumentNode;
     transformForLink(document: DocumentNode): DocumentNode;
-    readQuery<QueryType, TVariables = any>(options: DataProxy.Query<TVariables>, optimistic?: boolean): QueryType | null;
-    readFragment<FragmentType, TVariables = any>(options: DataProxy.Fragment<TVariables>, optimistic?: boolean): FragmentType | null;
+    readQuery<QueryType, TVariables = any>(options: DataProxy.Query<TVariables, QueryType>, optimistic?: boolean): QueryType | null;
+    readFragment<FragmentType, TVariables = any>(options: DataProxy.Fragment<TVariables, FragmentType>, optimistic?: boolean): FragmentType | null;
     writeQuery<TData = any, TVariables = any>(options: Cache.WriteQueryOptions<TData, TVariables>): void;
     writeFragment<TData = any, TVariables = any>(options: Cache.WriteFragmentOptions<TData, TVariables>): void;
     writeData<TData = any>({ id, data, }: Cache.WriteDataOptions<TData>): void;
diff --git a/node_modules/apollo-cache/lib/types/Cache.d.ts b/node_modules/apollo-cache/lib/types/Cache.d.ts
index f73aeb1..dfec0ef 100644
--- a/node_modules/apollo-cache/lib/types/Cache.d.ts
+++ b/node_modules/apollo-cache/lib/types/Cache.d.ts
@@ -4,12 +4,12 @@ export declare namespace Cache {
     interface EvictionResult {
         success: Boolean;
     }
-    interface ReadOptions<TVariables = any> extends DataProxy.Query<TVariables> {
+    interface ReadOptions<TVariables = any, TData = any> extends DataProxy.Query<TVariables, TData> {
         rootId?: string;
         previousResult?: any;
         optimistic: boolean;
     }
-    interface WriteOptions<TResult = any, TVariables = any> extends DataProxy.Query<TVariables> {
+    interface WriteOptions<TResult = any, TVariables = any> extends DataProxy.Query<TVariables, TResult> {
         dataId: string;
         result: TResult;
     }
diff --git a/node_modules/apollo-cache/lib/types/DataProxy.d.ts b/node_modules/apollo-cache/lib/types/DataProxy.d.ts
index 992df92..d3ff21a 100644
--- a/node_modules/apollo-cache/lib/types/DataProxy.d.ts
+++ b/node_modules/apollo-cache/lib/types/DataProxy.d.ts
@@ -1,19 +1,21 @@
 import { DocumentNode } from 'graphql';
+import { TypedDocumentNode } from '@graphql-typed-document-node/core';
+
 export declare namespace DataProxy {
-    interface Query<TVariables> {
-        query: DocumentNode;
+    interface Query<TVariables, TData = any> {
+        query: DocumentNode | TypedDocumentNode<TData, TVariables>;
         variables?: TVariables;
     }
-    interface Fragment<TVariables> {
+    interface Fragment<TVariables, TData = any> {
         id: string;
-        fragment: DocumentNode;
+        fragment: DocumentNode | TypedDocumentNode<TData, TVariables>;
         fragmentName?: string;
         variables?: TVariables;
     }
-    interface WriteQueryOptions<TData, TVariables> extends Query<TVariables> {
+    interface WriteQueryOptions<TData, TVariables> extends Query<TVariables, TData> {
         data: TData;
     }
-    interface WriteFragmentOptions<TData, TVariables> extends Fragment<TVariables> {
+    interface WriteFragmentOptions<TData, TVariables> extends Fragment<TVariables, TData> {
         data: TData;
     }
     interface WriteDataOptions<TData> {
@@ -26,8 +28,8 @@ export declare namespace DataProxy {
     };
 }
 export interface DataProxy {
-    readQuery<QueryType, TVariables = any>(options: DataProxy.Query<TVariables>, optimistic?: boolean): QueryType | null;
-    readFragment<FragmentType, TVariables = any>(options: DataProxy.Fragment<TVariables>, optimistic?: boolean): FragmentType | null;
+    readQuery<QueryType, TVariables = any>(options: DataProxy.Query<TVariables, QueryType>, optimistic?: boolean): QueryType | null;
+    readFragment<FragmentType, TVariables = any>(options: DataProxy.Fragment<TVariables, FragmentType>, optimistic?: boolean): FragmentType | null;
     writeQuery<TData = any, TVariables = any>(options: DataProxy.WriteQueryOptions<TData, TVariables>): void;
     writeFragment<TData = any, TVariables = any>(options: DataProxy.WriteFragmentOptions<TData, TVariables>): void;
     writeData<TData = any>(options: DataProxy.WriteDataOptions<TData>): void;
diff --git a/node_modules/apollo-cache/src/types/DataProxy.ts b/node_modules/apollo-cache/src/types/DataProxy.ts
index 8b85709..cc8a1bf 100644
--- a/node_modules/apollo-cache/src/types/DataProxy.ts
+++ b/node_modules/apollo-cache/src/types/DataProxy.ts
@@ -1,13 +1,14 @@
 import { DocumentNode } from 'graphql'; // eslint-disable-line import/no-extraneous-dependencies, import/no-unresolved
+import { TypedDocumentNode } from '@graphql-typed-document-node/core';
 
 export namespace DataProxy {
-  export interface Query<TVariables> {
+  export interface Query<TVariables, TData = any> {
     /**
      * The GraphQL query shape to be used constructed using the `gql` template
      * string tag from `graphql-tag`. The query will be used to determine the
      * shape of the data to be read.
      */
-    query: DocumentNode;
+    query: DocumentNode | TypedDocumentNode<TData, TVariables>;
 
     /**
      * Any variables that the GraphQL query may depend on.
@@ -15,7 +16,7 @@ export namespace DataProxy {
     variables?: TVariables;
   }
 
-  export interface Fragment<TVariables> {
+  export interface Fragment<TVariables, TData = any> {
     /**
      * The root id to be used. This id should take the same form as the
      * value returned by your `dataIdFromObject` function. If a value with your
@@ -29,7 +30,7 @@ export namespace DataProxy {
      * the shape of data to read. If you provide more than one fragment in this
      * document then you must also specify `fragmentName` to select a single.
      */
-    fragment: DocumentNode;
+    fragment: DocumentNode | TypedDocumentNode<TData, TVariables>;
 
     /**
      * The name of the fragment in your GraphQL document to be used. If you do
@@ -45,7 +46,7 @@ export namespace DataProxy {
   }
 
   export interface WriteQueryOptions<TData, TVariables>
-    extends Query<TVariables> {
+    extends Query<TVariables, TData> {
     /**
      * The data you will be writing to the store.
      */
@@ -53,7 +54,7 @@ export namespace DataProxy {
   }
 
   export interface WriteFragmentOptions<TData, TVariables>
-    extends Fragment<TVariables> {
+    extends Fragment<TVariables, TData> {
     /**
      * The data you will be writing to the store.
      */
@@ -87,7 +88,7 @@ export interface DataProxy {
    * Reads a GraphQL query from the root query id.
    */
   readQuery<QueryType, TVariables = any>(
-    options: DataProxy.Query<TVariables>,
+    options: DataProxy.Query<TVariables, QueryType>,
     optimistic?: boolean,
   ): QueryType | null;
 
@@ -97,7 +98,7 @@ export interface DataProxy {
    * provided to select the correct fragment.
    */
   readFragment<FragmentType, TVariables = any>(
-    options: DataProxy.Fragment<TVariables>,
+    options: DataProxy.Fragment<TVariables, FragmentType>,
     optimistic?: boolean,
   ): FragmentType | null;
 
