diff --git a/node_modules/graphql/execution/execute.d.ts b/node_modules/graphql/execution/execute.d.ts
index f927d90..fe4c23d 100644
--- a/node_modules/graphql/execution/execute.d.ts
+++ b/node_modules/graphql/execution/execute.d.ts
@@ -2,6 +2,7 @@ import { Maybe } from '../jsutils/Maybe';
 
 import { PromiseOrValue } from '../jsutils/PromiseOrValue';
 import { Path } from '../jsutils/Path';
+import { TypedDocumentNode } from "@graphql-typed-document-node/core";
 
 import { GraphQLError } from '../error/GraphQLError';
 
@@ -50,12 +51,12 @@ export interface ExecutionResult<TData = { [key: string]: any }> {
   data?: TData | null;
 }
 
-export interface ExecutionArgs {
+export interface ExecutionArgs<TData = any, TVariables = any> {
   schema: GraphQLSchema;
-  document: DocumentNode;
+  document: DocumentNode | TypedDocumentNode<TData, TVariables>;
   rootValue?: any;
   contextValue?: any;
-  variableValues?: Maybe<{ [key: string]: any }>;
+  variableValues?: Maybe<TVariables>;
   operationName?: Maybe<string>;
   fieldResolver?: Maybe<GraphQLFieldResolver<any, any>>;
   typeResolver?: Maybe<GraphQLTypeResolver<any, any>>;
@@ -73,17 +74,17 @@ export interface ExecutionArgs {
  *
  * Accepts either an object with named arguments, or individual arguments.
  */
-export function execute(args: ExecutionArgs): PromiseOrValue<ExecutionResult>;
-export function execute(
+export function execute<TData = any, TVariables = any>(args: ExecutionArgs<TData, TVariables>): PromiseOrValue<ExecutionResult<TData>>;
+export function execute<TData = any, TVariables = any>(
   schema: GraphQLSchema,
-  document: DocumentNode,
+  document: DocumentNode | TypedDocumentNode<TData, TVariables>,
   rootValue?: any,
   contextValue?: any,
-  variableValues?: Maybe<{ [key: string]: any }>,
+  variableValues?: Maybe<TVariables>,
   operationName?: Maybe<string>,
   fieldResolver?: Maybe<GraphQLFieldResolver<any, any>>,
   typeResolver?: Maybe<GraphQLTypeResolver<any, any>>,
-): PromiseOrValue<ExecutionResult>;
+): PromiseOrValue<ExecutionResult<TData>>;
 
 /**
  * Essential assertions before executing to provide developer feedback for
